### PR TITLE
'hybrid CRT' targets for the Windows platform

### PR DESCRIPTION
**Pull request by trueqbit**
Adds new 'hybrid CRT' targets for the Windows platform as requested in #20080.

### CODE REVIEW

**Comment by t8m** [comment]
Could you please sign CLA according to the https://www.openssl.org/policies/cla.html ? If your contribution wasn't done as part of work for any company, you can just submit ICLA and state that explicitly. Otherwise we would also need a CCLA signed by your company.

**Review by hlandau — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by trueqbit** [comment]
@paulidale If someone could revise the `remove_from_flags()` function, that would be great - Since I'm not a Perl programmer, I could only figure out how to pass it a "materialized" string, so calling `add(...)->(@_)` is currently necessary.

File: NOTES-WINDOWS.md [hunk 1]
```diff
@@ -77,6 +77,8 @@ Quick start
     - `perl Configure VC-WIN64A`    if you want 64-bit OpenSSL or
     - `perl Configure VC-WIN64-ARM` if you want Windows on Arm (win-arm64)
        OpenSSL or
+    - `perl Configure VC-WIN32-HYBRIDCRT`
+    - `perl Configure VC-WIN64A-HYBRIDCRT`
```
Comment (levitte) [inline NOTES-WINDOWS.md:81]: These lines should end with a short description, just like lines 77-78

**Comment by levitte** [comment]
> @paulidale If someone could revise the `remove_from_flags()` function, that would be great - Since I'm not a Perl programmer, I could only figure out how to pass it a "materialized" string, so calling `add(...)->(@_)` is currently necessary.

Coming up!

**Comment by trueqbit** [comment]
> > @paulidale If someone could revise the `remove_from_flags()` function, that would be great - Since I'm not a Perl programmer, I could only figure out how to pass it a "materialized" string, so calling `add(...)->(@_)` is currently necessary.
> 
> ~Coming up!~ Actually, upon a closer look, the way you do it is fine.

Thx! I just thought that the lazy evaluation machinery should be carried on.

**Comment by levitte** [comment]
> Thx! I just thought that the lazy evaluation machinery should be carried on.

Agreed, and I did try to make a simple change to `remove_from_flags`, but it turned out that because you add new flags at the same time, that became rapidly more complex than its worth spending time "perfecting".  I might take another stab at it later, though, as a separate PR.

**Review by hlandau — APPROVED** [review]

**Comment by t8m** [comment]
@paulidale please reconfirm

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks for the contribution.

**Pull request closed without merging**